### PR TITLE
LogManager: use layered config

### PR DESCRIPTION
**Pull request by ligfx**
Logical follow-up to #5764. Lets you control log configuration from the command line, which is cool.

The layered config code seems pretty easy to work with, though I'm not sure if I put stuff in the right places?

### CODE REVIEW

**Comment by ligfx** [comment]
@MerryMage

File: Source/Core/Common/Logging/LogManager.cpp [hunk 1]
```diff
@@ -10,15 +10,23 @@
...
 #include "Common/Logging/ConsoleListener.h"
 #include "Common/Logging/Log.h"
 #include "Common/Logging/LogManager.h"
 #include "Common/StringUtil.h"
 #include "Common/Timer.h"
+#include "Core/Config/Config.h"
 
 constexpr size_t MAX_MSGLEN = 1024;
 
+const Config::ConfigInfo<bool> LOGGER_WRITE_TO_FILE{
```
Comment (iwubcode) [inline Source/Core/Common/Logging/LogManager.cpp]: Not sure what everyone thinks of this but we ended up putting the video config options under `Source/Core/Core/Config` in their own files.  Does it make sense to do the same thing with the logging options?  It'd be nice to have the options all in one place.

Comment (ligfx) [inline Source/Core/Common/Logging/LogManager.cpp]: Yeah, that's the big question I have with this. The difference with log settings is that 1) they're all confined to this file anyways and 2) I didn't want to make settings for each log type. Is making config infos globally accessible (e.g. a "LoggerSettings.h") an architectural decision, or was it just a good fit for graphics settings?

**Comment by MerryMage** [comment]
LGTM.

**Pull request merged**